### PR TITLE
fix: use async api for email sending

### LINKED ISSUE #378: Async SMTP
The current implementation utilizes synchronous SMTP API which can lead to hangs of the whole server during password reset if SMTP server is not responding (timeout). It would be nice to use asynchronous version to make sure it won't block the main thread.

### PR DESCRIPTION
**Pull request by Michsior14**
Resolves #378

### CODE REVIEW

File: server/src/infra/mail.rs [hunk 1]
```diff
@@ -54,14 +54,15 @@ To reset your password please visit the following URL: {}/reset-password/step2/{
 Please contact an administrator if you did not initiate the process.",
         username, domain, token
     );
-    send_email(to, "[LLDAP] Password reset requested", body, options)
+    return send_email(to, "[LLDAP] Password reset requested", body, options).await;
```
Comment (nitnelave) [inline server/src/infra/mail.rs:57]: You don't need the `return`:
```rust
send_email(...).await
```

File: server/src/infra/mail.rs [hunk 2]
```diff
@@ -54,14 +54,15 @@ To reset your password please visit the following URL: {}/reset-password/step2/{
...
 }
 
-pub fn send_test_email(to: Mailbox, options: &MailOptions) -> Result<()> {
-    send_email(
+pub async fn send_test_email(to: Mailbox, options: &MailOptions) -> Result<()> {
+    return send_email(
```
Comment (nitnelave) [inline server/src/infra/mail.rs:61]: Same here, no return.

**Review by nitnelave — CHANGES_REQUESTED** [review]

**Comment by nitnelave** [comment]
I'm letting the checks run, but I think you'll need to run `cargo fmt`.

**Comment by Michsior14** [comment]
> I'm letting the checks run, but I think you'll need to run `cargo fmt`.

It should be already formatted correctly.

**Comment by Michsior14** [comment]
@nitnelave What about the clippy? Should I fix it in this PR or you prefer another one?

**Review by nitnelave — APPROVED** [review]

**Comment by nitnelave** [comment]
Thanks for the PR! Merged.

You can open another PR for the clippy fixes, if you want.

**Pull request merged**